src/components/Patient/PatientDetailsTab/Demography.tsx

```diff
@@ -61,11 +61,19 @@ export const Demography = (props: PatientProps) => {
     };
   }, [patientData.assigned_to_object]);
 
-  const { data: insuranceDetials } = useQuery(routes.hcx.policies.list, {
-    query: {
-      patient: id,
+  const { data: insuranceDetials, refetch } = useQuery(
+    routes.hcx.policies.list,
+    {
+      query: { patient: id },
+      prefetch: false, // Don't prefetch by default
     },
-  });
+  );
+
+  useEffect(() => {
+    if (patientData.id) {
+      refetch();
+    }
+  }, [patientData.id, refetch]);
```

Comment:
why disable prefetch? wouldn't `patientData.id` always be present?

Reply:
In `PatientHome.tsx`, the getPatient API fetches `patientData` and passes it to `Demography` as prop. 

Earlier, I disabled `prefetch`  for the insuranceDetails API in Demography to avoid calls with an undefined `patientData.id`. 

Now, by setting prefetch to !!patientData.id in the query itself, the API triggers only when the `patientData.id` is available,This prevents unnecessary call.

This change working correctly. Please confirm if this correct.

Reply:
you can get the patient id using the `useSlug` or by passing it as a prop right? We can prevent request waterfalls this way. There's no reason for fetching insurance details to wait until we get the patient object right?

Reply:
Ok got it .

Reply:
> you can get the patient id using the `useSlug` or by passing it as a prop right? We can prevent request waterfalls this way. There's no reason for fetching insurance details to wait until we get the patient object right?

Yes we are passing patient id as prop from PatientHome.tsx. This patient id is fetched from getPatient api request. 

But the tabs are rendered before the fetching is completed and props are passed as undefined to demography tab. 

So to prevent this i have made fetching insurance details to wait until we get the patient object.

Instead of this should i write a conditional statement in patienthome.tsx so that tabs are not rendered until patientdata is fetched

```javascript
{Tab && patientData?.id && (
              <Tab
                facilityId={facilityId || ""}
                id={patientData.id || ""}
                patientData={patientData}
                refetch={refetch}
              />
   )}

Reply:
We can get the patient ID from the URL right?

Reply:
Yes